src/routes/email/guards/email.guard.ts

```diff
@@ -0,0 +1,72 @@
+import { CanActivate, ExecutionContext, Inject, mixin } from '@nestjs/common';
+import { verifyMessage } from 'viem';
+import { LoggingService, ILoggingService } from '@/logging/logging.interface';
+
+export enum EmailGuardActionPrefix {
+  Register = 'email-register',
+  Delete = 'email-delete',
+}
+
+/**
+ * The EmailGuard guard should be used on routes that require authenticated
+ * actions for email addresses modification.
+ *
+ * This guard therefore validates if the message came from the specified account.
+ *
+ * The following message should be signed:
+ * ${EmailGuardActionPrefix}-${chainId}-${safe}-${emailAddress}-${account}-${timestamp}
+ *
+ * (where ${} represents placeholder values for the respective data)
+ *
+ * To use this guard, 'EmailGuardActionPrefix' should be specific and the route should have:
+ * - the 'chainId' declared as a parameter
+ * - the 'safeAddress' declared as a parameter
+ * - the 'emailAddress' as part of the JSON body (top level)
+ * - the 'account' as part of the JSON body (top level)
+ * - the 'signature' as part of the JSON body (top level) - see message format to be signed
+ * - the 'timestamp' as part of the JSON body (top level)
+ */
+export const EmailGuard = (prefix: EmailGuardActionPrefix) => {
+  class EmailRegistrationGuardMixin implements CanActivate {
+    constructor(
+      @Inject(LoggingService) readonly loggingService: ILoggingService,
+    ) {}
+
+    async canActivate(context: ExecutionContext): Promise<boolean> {
+      const request = context.switchToHttp().getRequest();
+
+      const chainId = request.params['chainId'];
+      const safe = request.params['safeAddress'];
+      const emailAddress = request.body['emailAddress'];
```

Comment:
The deletion shouldn't require an `emailAddress` in the body right?

Reply:
Indeed. For simplicity, I reverted the `mixin` change and created a separate guard for deletion in e7f957d61d1d799891ab901acb8fc1bd750d1810.